File: 03- Subprogramas y recursividad/subprogramas/java/excepciones/ValidarEdad.java
/**
 * Descripción: Hace uso de una excepción personalizada para validar las edades inválidas.
 * Autor: Santiago Nicolás De la mora Núñez
 * Fecha: 25/02/2026
 */

public class ValidarEdad {
    public static void main(String[] args) {
        try {
            validarEdad(15);
        } catch (EdadInvalidaException e) {
            System.err.println("Error: " + e.getMessage());
        }
    }

    public static void validarEdad(int edad) throws EdadInvalidaException {
        if (edad < 18) {
            throw new EdadInvalidaException("Debe de ser mayor de edad.");
        }
    }
}

File: 03- Subprogramas y recursividad/subprogramas/python/excepciones/validar_edad.py
"""
Descripción: Hace uso de una excepción personalizada para validar las edades inválidas.
Autor: Santiago Nicolás De la mora Núñez
Fecha de creación: 10/04/2026
"""

from edad_invalida_exception import edad_invalida_exception

def main():
    try:
        validar_edad(15)
    except edad_invalida_exception as e:
        print(f"Error: {e}")

def validar_edad(edad):
    if edad < 18:
        raise edad_invalida_exception("Debe de ser mayor de edad.")

# Ejecuta la aplicación
if __name__ == '__main__':
    main()